pyomo/contrib/solver/base.py

```diff
@@ -435,9 +466,13 @@ def _map_results(self, model, results):
         ]
         legacy_soln.status = legacy_solution_status_map[results.solution_status]
         legacy_results.solver.termination_message = str(results.termination_condition)
-        legacy_results.problem.number_of_constraints = model.nconstraints()
-        legacy_results.problem.number_of_variables = model.nvariables()
-        number_of_objectives = model.nobjectives()
+        legacy_results.problem.number_of_constraints = len(
+            list(model.component_map(ctype=Constraint))
+        )
+        legacy_results.problem.number_of_variables = len(
+            list(model.component_map(ctype=Var))
+        )
+        number_of_objectives = len(list(model.component_map(ctype=Objective)))
```

Comment:
These are almost certainly *not* the correct numbers (you are returning the number of component containers on the root block, and not the number of ComponentData objects on the entire tree).  Worse, tabulating "less incorrect" information is going to be expensive.  Do we really need to provide this in the results, or for the LegacySolverWrapper can we just set them to `NaN`?

Reply:
I don't mind setting them to `NaN` but that's up to you all. Is this something that is promised in the book? Also, it is printed out when you run `model.pprint()` which is why I put this in here to begin with.

Reply:
Just speaking personally, I generally would not look to the solver results object for number of variables, constraints and objectives. I would either look at my original model (for the full model) or the solver logs (for what the solver saw).

That said, this was before we had a linear presolve, which can change the number of variables and constraints.

Reply:
So I am... not sure how to handle this now. Suggestion?